src/main/kotlin/se/uulm/snowballr/backend/service/ProjectService.kt

```diff
@@ -91,8 +95,27 @@ class ProjectService(
         return repo.getProjectById(projectId).toGrpcProject()
     }
 
-    override suspend fun createProject(request: GrpcProject.Create): GrpcProject =
-        repo.createProject(request, GrpcContext.getUserIdFromContext()).toGrpcProject()
+    override suspend fun createProject(request: GrpcProject.Create): GrpcProject {
+        val currentUser = userRepo.getUserById(GrpcContext.getUserIdFromContext())
+        val userSettings = userRepo.getUserSettings(currentUser.id)
+        val userDefaultCriteria = criterionRepo.getCriteriaByIds(userSettings.criteriaIds)
+
+        val project = repo.createProject(request, GrpcContext.getUserIdFromContext(), userSettings).toGrpcProject()
+
+        for (criterion in userDefaultCriteria) {
+            val criterionRequest = CriterionOuterClass.Criterion.Create
+                .newBuilder()
+                .setTag(criterion.tag)
+                .setName(criterion.name)
+                .setDescription(criterion.description)
+                .setCategory(criterion.category)
+                .setProjectId(project.id ?: "")
```

Comment:
Do we require this null fallback (elvis operator)? The following should be just fine.
```suggestion
                .setProjectId(project.id)
```

Reply:
If we don't add it, we get a linter error. So either we let it as it is or we add a suppress linter annotation. I don't care.

Reply:
Ah, I didn't check the linter. This is because the gRPC code is in Java, and therefore in Kotlin every property is assumed to be nullable.
We don't need to convert the project to a gRPC object until the end of this method, so we could just use the DTO for this and then call `.setProjectId(project.id.toString())` and at the end `return project.toGrpcProject()`.

Reply:
You are right. Could have figured this out myself... Thank you.